### PR TITLE
Nft holder creation

### PR DESCRIPTION
**Pull request by abhishek-k-sharma-lab49**

### CODE REVIEW

File: .github/workflows/e2e-test.yml [hunk 1]
```diff
@@ -27,7 +27,6 @@ jobs:
   suite-1:
```
Comment (amandeeprehal-lab49) [inline .github/workflows/e2e-test.yml:27]: No change pls remove this file.

File: contracts/common/BaseHTS.sol [hunk 2]
```diff
@@ -28,6 +28,21 @@ contract BaseHTS is HederaTokenService, IBaseHTS {
             );
     }
 
+    function transferNFTPublic(
+        address token,
+        address sender,
+        address receiver,
+        int256 serial
```
Comment (amandeeprehal-lab49) [inline contracts/common/BaseHTS.sol:35]: We can keep it int64.

Comment (amandeeprehal-lab49) [reply contracts/common/BaseHTS.sol:35]: `serial`

File: contracts/governance/TokenHolder.sol [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    IBaseHTS internal _tokenService;
+    address internal _token;
+
+    function initialize(
+        IBaseHTS tokenService,
+        address token
+    ) public initializer {
+        _tokenService = tokenService;
+        _token = token;
+        _associateToken(address(this), address(_token));
```
Comment (amandeeprehal-lab49) [inline contracts/governance/TokenHolder.sol:20]: This was the line you were looking for!

File: contracts/governance/TokenHolder.sol [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    function _associateToken(address account, address token) private {
+        (bool success, ) = address(_tokenService).delegatecall(
```
[the remark this answers is not in the document]
Comment (amandeeprehal-lab49) [reply contracts/governance/TokenHolder.sol:24]: if. so, pls add this line 

```
 /// @custom:oz-upgrades-unsafe-allow delegatecall
```

on line 22 to silent the warning.

File: test/nft-holder-test.ts [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    const tokenId = await nftHolder.balanceOfVoter(signers[0].address);
+    expect(tokenId).to.be.equal(0);
+    await nftHolder.grabTokensFromUser(signers[0].address, 1);
+    const tokenIdAfterGrab = await nftHolder.balanceOfVoter(signers[0].address);
+    expect(tokenIdAfterGrab).to.be.equal(1);
+  });
+
+  it("Verify NFTHolder grabtoken pass", async function () {
+    const { nftHolder, signers, tokenCont } = await loadFixture(deployFixture);
+    const userTokens = await tokenCont.balanceOf(signers[0].address);
```
Comment (amandeeprehal-lab49) [inline test/nft-holder-test.ts:73]: pls assert `userTokens` as a pre condition

File: test/nft-holder-test.ts [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    const contractTokens = await tokenCont.balanceOf(nftHolder.address);
+    const userTokens1 = await tokenCont.balanceOf(signers[0].address);
+    expect(userTokens1).to.be.equal(0);
+    expect(userTokens).to.be.equal(contractTokens);
+  });
+
+  it("Verify Add and remove active proposals", async function () {
+    const { nftHolder, signers } = await loadFixture(deployFixture);
+    await nftHolder.addProposalForVoter(signers[0].address, 1);
+    await nftHolder.addProposalForVoter(signers[1].address, 1);
```
Comment (amandeeprehal-lab49) [inline test/nft-holder-test.ts:84]: I think, you dont need this line.

File: test/nft-holder-test.ts [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+    await nftHolder.addProposalForVoter(signers[0].address, 1);
+    await expect(nftHolder.revertTokensForVoter()).to.revertedWith(
+      "User's Proposals are active"
+    );
+  });
+
+  it("Verify NFTHolder revertTokensForVoter pass", async function () {
+    const { nftHolder, signers, tokenCont } = await loadFixture(deployFixture);
+    nftHolder.grabTokensFromUser(signers[0].address, 2);
+    const response = await nftHolder.callStatic.revertTokensForVoter();
```
Comment (amandeeprehal-lab49) [inline test/nft-holder-test.ts:121]: pls verify the balances as well both for user and contract.

File: .github/workflows/e2e-test.yml (hunk 1 above)
Comment (abhishek-k-sharma-lab49) [reply .github/workflows/e2e-test.yml]: This file adds an extra line at the end of the file, on commit.

File: test/nft-holder-test.ts [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+    const tokenId = await nftHolder.balanceOfVoter(signers[0].address);
+    expect(tokenId).to.be.equal(0);
+    await nftHolder.grabTokensFromUser(signers[0].address, 1);
+    const tokenIdAfterGrab = await nftHolder.balanceOfVoter(signers[0].address);
+    expect(tokenIdAfterGrab).to.be.equal(1);
+  });
+
+  it("Verify NFTHolder grabtoken pass", async function () {
+    const { nftHolder, signers, tokenCont } = await loadFixture(deployFixture);
+    await nftHolder.grabTokensFromUser(signers[0].address, tokenSerial);
```
Comment (amandeeprehal-lab49) [inline test/nft-holder-test.ts:77]: Pls assert user balance before calling garb

**Pull request merged**